tavern/util/dict_util.py

```diff
@@ -301,12 +301,28 @@ def _format_err(which):
                 except KeyError:
                     logger.debug("Skipping comparing missing key %s due to strict=%s", key, strict)
         elif isinstance(expected_val, list):
-            if len(expected_val) != len(actual_val):
-                raise_from(exceptions.KeyMismatchError("Length of returned list was different than expected - expected {} items, got {} ({})".format(len(expected_val), len(actual_val), full_err())), e)
-
             # TODO
             # Check things in the wrong order?
 
+            if not strict:
+                # Iterate over list items to see if any of them match...
+                for i, e_val in enumerate(expected_val):
+                    for a_val in actual_val:
+                        try:
+                            check_keys_match_recursive(e_val, a_val, keys + [i], strict)
+                        except exceptions.KeyMismatchError:
+                            pass
+                        else:
+                            logger.debug("One (or more) list items present - continuing due to strict=%s", strict)
+                            # TODO
+                            # do we need to check more here?
```

Comment:
What were you thinking here?

Reply:
:man_shrugging: I think I wrote that a while ago, it can be removed now I think